### PR TITLE
refactor OIDMatcher into separate utils package

### PR DESCRIPTION
**Pull request by linus-sun**
#### Summary

This PR refactors the OIDMatcher type into a separate utils package, which will subsequently enable [this PR](https://github.com/sigstore/rekor-monitor/pull/446) to refactor the renderFulcioOIDMatchers function into a struct method (re: [discussion here](https://github.com/sigstore/rekor-monitor/pull/446/files#r1755754767)). 

#### Release Note

NONE

#### Documentation

NONE- related documentation changes will happen in their respective PRs.

### CODE REVIEW

**Review by mihaimaruseac — COMMENTED** [review]
This looks good, but question on pkg/util/file/file.go and pkg/util/file/file_test.go as they seem unrelated: do we want them in this PR?

**Comment by linus-sun** [comment]
> This looks good, but question on pkg/util/file/file.go and pkg/util/file/file_test.go as they seem unrelated: do we want them in this PR?

I believe they're referenced because I moved them out of the `pkg/util` folder itself into a separate `pkg/util/file` folder so I can also put `identity_utils` in its own `util` subfolder, otherwise I'd have to put `identity_utils` in the `file` package, which doesn't seem suitable

**Comment by linus-sun** [comment]
Just realized this duplicates the file utils, fixing that now

File: cmd/verifier/main.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 	"time"
 
 	"github.com/sigstore/rekor-monitor/pkg/rekor"
-	file "github.com/sigstore/rekor-monitor/pkg/util"
+	file "github.com/sigstore/rekor-monitor/pkg/util/file"
```
Comment (haydentherapper) [inline cmd/verifier/main.go:30]: nit, you can remove `file` from the beginning of this declaration as this package is now named `file`

File: pkg/util/identity/identity.go
Comment (haydentherapper) [inline pkg/util/identity/identity.go:1]: WDYT about putting this under `pkg/identity`? I think we can expand an identity package over time as we reorganize this codebase, and avoid over-expanding utilities.

**Review by mihaimaruseac — DISMISSED** [review]

File: pkg/util/identity/identity.go
Comment (linus-sun) [reply pkg/util/identity/identity.go:1]: that makes sense, i'll keep that in mind as i continue looking through the codebase

**Review by mihaimaruseac — APPROVED** [review]

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**